src/jormungandr/explorer/src/logging.rs

```diff
@@ -129,66 +129,83 @@ impl LogSettings {
                         cause,
                     })?;
                 let (non_blocking, guard) = tracing_appender::non_blocking(file);
-                guards.push(guard);
 
                 self.init_subscriber(non_blocking);
+
+                guard
             }
-            #[cfg(feature = "systemd")]
-            LogOutput::Journald => {
-                self.config.format.require_default()?;
-                let layer = tracing_journald::layer().map_err(Error::Journald)?;
-                tracing_subscriber::registry()
-                    .with(self.config.level)
-                    .with(layer)
-                    .init();
-            }
-            #[cfg(feature = "gelf")]
-            LogOutput::Gelf { backend, .. } => {
-                let (layer, task) = tracing_gelf::Logger::builder()
-                    .connect_tcp(backend)
-                    .map_err(Error::Gelf)?;
-                tokio::spawn(task);
-                tracing_subscriber::registry()
-                    .with(self.config.level)
-                    .with(layer)
-                    .init();
-            }
+        };
+
+        let default_settings = Self::default();
+
+        if self.output != default_settings.output {
+            tracing::info!(
+                "log output overriden from command line: {:?} replaced with {:?}",
+                default_settings.output,
+                self.output
+            );
+        }
+        if self.level != default_settings.level {
+            tracing::info!(
+                "log level overriden from command line: {:?} replaced with {:?}",
+                default_settings.level,
+                self.level
+            );
+        }
+        if self.format != default_settings.format {
+            tracing::info!(
+                "log format overriden from command line: {:?} replaced with {:?}",
+                default_settings.format,
+                self.format
+            );
         }
 
-        Ok((guards, self.msgs))
+        Ok(LogGuard {
+            _nonblocking_worker_guard: nonblocking_worker_guard,
+        })
     }
 
-    fn init_subscriber(&self, non_blocking: tracing_appender::non_blocking::NonBlocking) {
-        match self.config.format {
-            LogFormat::Default | LogFormat::Plain => {
+    fn init_subscriber(
```

Comment:
does this subscriber have global scope? just wondering how we can log everything in the explorer?

Reply:
Yes. the `.init()` calls on the tracing subscriber `Registry` tries to set it as the global subscriber (https://docs.rs/tracing-subscriber/latest/tracing_subscriber/registry/struct.Registry.html#method.init).